exec/java-exec/src/main/resources/rest/profile/profile.ftl

```diff
@@ -506,7 +504,8 @@ table.sortable thead .sorting_desc { background-image: url("/static/img/black-de
     document.getElementById('queryForm')
             .addEventListener('keydown', function(e) {
       if (!(e.keyCode == 13 && (e.metaKey || e.ctrlKey))) return;
-      if (e.target.form) doSubmitQueryWithUserName();
+      if (e.target.form) 
+        <#if model.isImpersonationOnlyEnabled()>doSubmitQueryWithUserName()<#else>submitQuery()</#if>;
```

Comment:
I believe the root cause of the issue is here and below where for non-impersonation case submit was not actually done. In those cases username will be null inside `doSubmitQueryWithUserName` and will return with an alert.

Reply:
Yes, this was the root cause, because we trap the key even (Ctrl + Enter) and submit the query. 
We were trapping the event it for only impersonation scenarios. This takes care of that and abstracts the submit call. (i.e. the AJAX call). This way, we have only one method to do the actual submission, and the non-existent `username` will be handled by not setting header.

Also, PR #1592 will use this method to further improve the user experience by blocking interaction.